### ISSUE TITLE
Add Full-text Search for Abstracts and (potentially) Body

### ISSUE THREAD

**Issue by adelavega** [neurostuff/neurostore#473]
First, on the data side, we need to ingest Abstract into the database. 

Second, Postgres comes with Full-text Search capabilities.

See these articles:

https://amitosh.medium.com/full-text-search-fts-with-postgresql-and-sqlalchemy-edc436330a0c
https://www.crunchydata.com/blog/postgres-full-text-search-a-search-engine-in-a-database
https://stackoverflow.com/questions/13837111/tsvector-in-sqlalchemy

In short, postgres pre-computes versions of the text which are normalized, and input search queries are also normalized. 

This means that even if there is not a perfect text match, you can still get a result. 

I would suggest we made this the default search method, and only allow users to select fields in Advanced Search.

**Comment by adelavega** [comment]
We might have to define a custom dictionary to include terms outside of the English dictionary built into postgres

**Comment by adelavega** [comment]
We also need to decide if search will by default be boolean (i..e 'AND' between tokens), or full phrase matching (i.e. in order). We could allow both potentially?

**Comment by adelavega** [comment]
To automatically convert phrases we can use: "[phraseto_tsquery](https://www.postgresql.org/docs/current/textsearch-controls.html#TEXTSEARCH-PARSING-QUERIES)"

**Comment by adelavega** [comment]
Postgres even supports showing search highlights (https://www.postgresql.org/docs/current/textsearch-controls.html#TEXTSEARCH-HEADLINE) so this looks like a complete solution for our needs.

**Comment by adelavega** [comment]
Important: use `coalesce` to make a multi-index (i.e. title, abstract, body), and still create an index if one of those is null:

https://www.postgresql.org/docs/current/textsearch-tables.html
